### PR TITLE
Add an example on how to create XYZ or ArcGISRest sources

### PR DESCRIPTION
**Pull request by bartvde**
from the capabilities JSON response

### CODE REVIEW

**Comment by bartvde** [comment]
Tested with the following services:

http://tiles.arcgis.com/tiles/nSZVuSZjHpEZZbRo/arcgis/rest/services/TMK1850/MapServer
http://services.arcgisonline.com/arcgis/rest/services/ESRI_Imagery_World_2D/MapServer
http://tiles.arcgis.com/tiles/nGt4QxSblgDfeJn9/arcgis/rest/services/Milan_Population_Density_6class/MapServer

File: examples/arcgis-rest-capabilities.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  var sourceInfo;
+  if (config.tileInfo) {
+    sourceInfo = createXYZSource(config, serviceUrl, attributions, projection,
+        resolutions);
+  } else {
+    sourceInfo = createArcGISRestSource(attributions, serviceUrl);
+  }
+  return new ol.layer.Tile({
+    source: sourceInfo.source,
+    extent: sourceInfo.extent
```
Comment (ahocevar) [inline examples/arcgis-rest-capabilities.js]: I don't think it's good to restrict the extent on the layer level. Should be done on the tilegrid level instead IMHO.

**Comment by ahocevar** [comment]
Why not create an `ol.source.TileArcGISRest.optionsFromCapabilities` function with a proper tile grid configuration? Then the application code could be much simpler.

**Comment by bartvde** [comment]
@ahocevar the code will create an XYZ or ArcGISRest source, or you don't think that is an issue here? Actually doing the config for the XYZ is 95% of this code.

**Comment by ahocevar** [comment]
Ah, I see now. Still I think that's a lot of application code. Not sure where to fit it in the library though.

**Comment by bartvde** [comment]
I agree, but I also couldn't find a good spot in the library.

**Comment by bartvde** [comment]
Unless we create a new source (for example ol.source.ArcGISXYZ) that has a optionsFromCapabilities function.

**Comment by ahocevar** [comment]
Ok, to bring this into the library we'll need an externs file and probably a format. So maybe just keep it like it is, but change the code so the extent is restricted through the tile grid, not the layer.

**Comment by bartvde** [comment]
Okay it might make sense to create a separate library for ol3-esri integration, which will also contain the style conversion that I need to work on soon. We can then showcase these in an example will less code. So I'll hold off for this for now.

**Comment by bartvde** [comment]
Started on the external lib idea here: https://github.com/bartvde/ol3-esri
Updated the branch to show usage.
WIP.

**Comment by bartvde** [comment]
I'm closing this since the example will be in ```ol3-esri``` instead

**Pull request closed without merging**